lib/xiaomi.js

```diff
@@ -160,7 +160,7 @@ const numericAttributes2Payload = (msg, meta, model, options, dataObject) => {
             payload.mode_switch = {4: 'anti_flicker_mode', 1: 'quick_mode'}[value];
             break;
         case '5':
-            if (['JT-BZ-01AQ/A', 'RTCZCGQ11LM'].includes(model.model)) {
+            if (['JT-BZ-01AQ/A', 'LLKZMK11LM', 'RTCZCGQ11LM', 'SSM-U01'].includes(model.model)) {
```

Comment:
I'm pretty sure the major part (if not all) the devices of Xiaomi with a field 5 use it for power_outage_count. Maybe is better to simply remove the condition and let it for all waiting for some complains if any?

Reply:
As far as I noticed, this parameter applies only to devices with stationary power supply, so if you remove the filter by models, then it seems to me that you need to add a filter by power source.

Reply:
Not exactly, only if the device publishes a "5" attribute. Do you know if some battery device publishes it?

Reply:
> Not exactly, only if the device publishes a "5" attribute.

Of course, only if the device publishes a "5" attribute.

> Do you know if some battery device publishes it?

So far I haven't come across any.

Reply:
> Do you know if some battery device publishes it?

I'm sorry, but my previous answer is not correct, because I didn't understand the question correctly.

I have 3 battery-powered devices that publishes a "5" attribute, but at the same time it is clearly not the value of _power_outage_count_, but something else:

- **[WSDCGQ12LM](https://www.zigbee2mqtt.io/devices/WSDCGQ12LM.html)**

  ```javascript
  lumi.sensor_ht.agl02: unknown index 1 with value 3035
  lumi.sensor_ht.agl02: unknown index 3 with value 22
  lumi.sensor_ht.agl02: unknown index 4 with value 424
  lumi.sensor_ht.agl02: unknown index 5 with value 36
  lumi.sensor_ht.agl02: unknown index 6 with value 11
  lumi.sensor_ht.agl02: unknown index 8 with value 284
  lumi.sensor_ht.agl02: unknown index 10 with value 0
  lumi.sensor_ht.agl02: unknown index 12 with value 1
  lumi.sensor_ht.agl02: unknown index 100 with value 2270
  lumi.sensor_ht.agl02: unknown index 101 with value 2098
  lumi.sensor_ht.agl02: unknown index 102 with value 1001
  ```
- **[RTCGQ12LM](https://www.zigbee2mqtt.io/devices/RTCGQ12LM.html)**

  ```javascript
  lumi.motion.agl02: unknown index 1 with value 3060
  lumi.motion.agl02: unknown index 3 with value 25
  lumi.motion.agl02: unknown index 4 with value 424
  lumi.motion.agl02: unknown index 5 with value 776
  lumi.motion.agl02: unknown index 6 with value 19
  lumi.motion.agl02: unknown index 8 with value 291
  lumi.motion.agl02: unknown index 10 with value 0
  lumi.motion.agl02: unknown index 12 with value 1
  lumi.motion.agl02: unknown index 100 with value 0
  lumi.motion.agl02: unknown index 101 with value 2
  lumi.motion.agl02: unknown index 102 with value 3
  lumi.motion.agl02: unknown index 103 with value 1
  lumi.motion.agl02: unknown index 104 with value 168
  ```
- **[RTCGQ13LM](https://www.zigbee2mqtt.io/devices/RTCGQ13LM.html)**

  ```javascript
  lumi.motion.agl04: unknown index 1 with value 3112
  lumi.motion.agl04: unknown index 3 with value 25
  lumi.motion.agl04: unknown index 4 with value 424
  lumi.motion.agl04: unknown index 5 with value 491
  lumi.motion.agl04: unknown index 6 with value 15
  lumi.motion.agl04: unknown index 8 with value 282
  lumi.motion.agl04: unknown index 10 with value 0
  lumi.motion.agl04: unknown index 12 with value 1
  lumi.motion.agl04: unknown index 100 with value 0
  lumi.motion.agl04: unknown index 101 with value 0
  lumi.motion.agl04: unknown index 102 with value 3
  lumi.motion.agl04: unknown index 103 with value 1
  lumi.motion.agl04: unknown index 104 with value 168
  lumi.motion.agl04: unknown index 105 with value 2
  ```

Reply:
Yes, you have reason. They are too high to be the times you changed the battery... We can left the filter by model or as you say, as alternative, try what happens filtering by power source.

Reply:
Instead of doing a model based check, we can check the power source: `if (meta.device.powerSource === 'Mains (single phase)')`

Reply:
Ok, modified to use the power source. Tested and it works for my devices.

Reply:
@Koenkk don't merge this. It seems is not working. Wait please to let me do more tests.

Reply:
Ok will wait, can you also add `power_outage_count` expose to more devices?

Reply:
Fixed, the correct one was `if (meta.device.powerSource === 'DC Source')`.

I'm sorry but I don't have more devices with DC Source to test them. But it you or @Otnow can point me to some of them I will add it. My idea was to wait to reports of "state" with the value to add them in the future.

Reply:
I suspect that the devices that use the aqara_opple converter will have it. But I can confirm.

Reply:
@Koenkk It seems some devices use the `DC Source` like [LLKZMK11LM](https://www.zigbee2mqtt.io/devices/LLKZMK11LM.html#xiaomi-llkzmk11lm) and others the `Mains (single phase)` like the [SSM-U01](https://www.zigbee2mqtt.io/devices/SSM-U01.html#xiaomi-ssm-u01). I have let both finally.